### DISCUSSION TITLE
Rule hits 3 times - bug or intention ?

### DISCUSSION THREAD

**Discussion by NiklausS** [arendst/Tasmota#20226] [Q&A] [answered]
Hi all,
just stumbled on this:

```
Rule1
  ON BMP280#Temperature<27.3   DO Backlog   Var3 = %value% ENDON  
```  
this rule 'hits' 3 times every second.
```
11:03:17.022 RUL: BMP280#TEMPERATURE<27.3 performs "Backlog   Var3 = 21.7"
11:03:17.033 RSL: RESULT = {"Var3":"21.700"}
11:03:17.931 RUL: BMP280#TEMPERATURE<27.3 performs "Backlog   Var3 = 21.7"
11:03:17.942 RSL: RESULT = {"Var3":"21.700"}
11:03:18.835 RUL: BMP280#TEMPERATURE<27.3 performs "Backlog   Var3 = 21.7"
11:03:18.847 RSL: RESULT = {"Var3":"21.700"}
11:03:19.722 RUL: BMP280#TEMPERATURE<27.3 performs "Backlog   Var3 = 21.7"
11:03:19.734 RSL: RESULT = {"Var3":"21.700"}

```
same if i do it in berry:
```
def rule_temp(value)
  print("The temp is bright", str(value))
  # bmp280.set_text("Temp=" + str(value) + "°")
end
tasmota.add_rule("BMP280#Temperature<27.3", rule_temp )
```
```
The temp is bright 21.7
The temp is bright 21.7
The temp is bright 21.7
```
everything is fine, but i don't see the point that this rule hits every second 3 times - i have some extensive rules where i have not noticed this behavior in older Tasmota installations.

i'm on a 13.3.0(lvgl-haspmota) with a black M5Stack 

Regards
 Nik

**Comment by sfromis** [comment, marked as answer]
That is entirely normal. You get the temperature value when it is updated every cycle, changed or not. If you want it at a slower pace, then it depends on when you specifically want to process values.

What you get with other rules depends on how your'e using what features there.

**Comment by NiklausS** [comment]
Thank you very much for the quick answer.

When looking at the serial out this is matches perfectly - 1/sec
However, as the webconsole seems to update only every 2+ seconds i get there 2-3 triggers per update

Nik

**Reply by sfromis** [reply]
By default the web UI (not specifically console) only receives updates every 2.345 seconds, unless you change this config setting.
